github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverter.java

```diff
@@ -42,12 +42,18 @@ public static PullRequestToCauseConverter toGitHubPRCause(GitHubPRRepository loc
         return new PullRequestToCauseConverter(localRepo, listener, trigger);
     }
 
+    /**
+     * @return only real trigger cause if matched trigger (not skip) event found for this remotePr.
+     */
     @Override
     public GitHubPRCause apply(final GHPullRequest remotePR) {
-        return from(trigger.getEvents())
+        final GitHubPRCause gitHubPRCause = from(trigger.getEvents())
                 .transform(toCause(remotePR))
                 .filter(notNull())
-                .firstMatch(new SkippedCauseFilter(listener)).orNull();
+                .first()
+                .get();
+
+        return gitHubPRCause.isSkip() ? null : gitHubPRCause;
```

Comment:
> In TODO migrate to java8. Do you know whether it has suitable methods?

Yes it does. See [Stream.reduce](https://docs.oracle.com/javase/tutorial/collections/streams/reduction.html#reduce) documentation.